common/reconciliation/fetcher/timer_test.go

```diff
@@ -0,0 +1,117 @@
+// The MIT License (MIT)
+//
+// Copyright (c) 2017-2020 Uber Technologies Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in all
+// copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+// SOFTWARE.
+
+package fetcher
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	"github.com/golang/mock/gomock"
+	"github.com/stretchr/testify/require"
+	"github.com/uber/cadence/common/pagination"
+	"github.com/uber/cadence/common/persistence"
+	"github.com/uber/cadence/common/reconciliation/entity"
+)
+
+func TestGetUserTimers(t *testing.T) {
+	ctrl := gomock.NewController(t)
+	defer ctrl.Finish()
+
+	mockRetryer := persistence.NewMockRetryer(ctrl)
+	ctx := context.Background()
+	pageSize := 10
+	minTimestamp := time.Now().Add(-time.Hour)
+	maxTimestamp := time.Now()
+
+	testCases := []struct {
+		name          string
+		setupMock     func() []*persistence.TimerTaskInfo
+		expectedPage  pagination.Page
+		expectedError bool
+	}{
+		{
+			name: "Success",
```

Comment:
great to have at least happy path covered but the failure cases are more interesting. I'd add following cases
1. pass non-nil token and validate it's passed down to `pr.GetTimerIndexTasks`
2. response has other timer types that are skipped ([this condition](https://github.com/taylanisikdemir/cadence/blob/e13da5812c24f55b7156c9d9b02bb94f7d5512b6/common/reconciliation/fetcher/timer.go#L69))
3. invalid timer causes error to be returned ([this condition](https://github.com/taylanisikdemir/cadence/blob/e13da5812c24f55b7156c9d9b02bb94f7d5512b6/common/reconciliation/fetcher/timer.go#L83))